### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Vasyl-Poremchuk**

### CODE REVIEW

File: cinema/serializers.py [hunk 1]
```diff
@@ -1 +1,106 @@
...
+        slug_field="name",
+    )
+    actors = serializers.SlugRelatedField(
+        many=True,
+        read_only=True,
+        slug_field="full_name",
+    )
+
+    class Meta:
+        model = Movie
```
Comment (wQuelS) [inline cinema/serializers.py:62]: Your MovieListSerializer inherits from MovieSerializer. Do you really need class Meta here with exact same fields ?

File: cinema/views.py [hunk 2]
```diff
@@ -1 +1,81 @@
...
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action == "list":
```
Comment (wQuelS) [inline cinema/views.py:45]: Please check for similar or duplicate queries with debug_toolbar when you retrieving detail info about movie

File: cinema/views.py [hunk 3]
```diff
@@ -1 +1,81 @@
...
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+    serializer_class = MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action == "list":
```
Comment (wQuelS) [inline cinema/views.py:68]: same, what about N+1 problem when retrieving movie_sessions detail info?

**Review by wQuelS — CHANGES_REQUESTED** [review]

**Review by wQuelS — APPROVED** [review]
Well done!

File: cinema/models.py [hunk 4]
```diff
@@ -27,6 +27,10 @@ class Actor(models.Model):
     first_name = models.CharField(max_length=255)
     last_name = models.CharField(max_length=255)
 
+    @property
+    def full_name(self):
+        return f"{self.first_name} {self.last_name}"
+
     def __str__(self):
         return self.first_name + " " + self.last_name
```
Comment (Nattalli) [inline cinema/models.py:35]: 1) add a type annotation
2) These two methods do the same thing. Maybe it's better to use one of them in the other?

File: cinema/serializers.py [hunk 5]
```diff
@@ -1 +1,98 @@
...
+    MovieSession,
+)
+
+
+class ActorSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Actor
+        fields = (
+            "id", "first_name",
+            "last_name", "full_name",
```
Comment (Nattalli) [inline cinema/serializers.py:17]: hmm, did you run `black`? It doesn't look good to me, but I can't tell that it's a mistake

Comment (Nattalli) [reply cinema/serializers.py:17]: just maybe 
1) 
```python
fields = ("a", "b", "c", "d")
```
or 
2) 
```python
fields = (
"a",
"b",
"c",
"d"
)
```
would that be better? WDYT?

**Review by Nattalli — CHANGES_REQUESTED** [review]

File: cinema/models.py (hunk 4 above)
Comment (Vasyl-Poremchuk) [reply cinema/models.py:35]: Oh, I didn't notice it :(

File: cinema/serializers.py (hunk 5 above)
Comment (Vasyl-Poremchuk) [reply cinema/serializers.py:17]: I agree with you, the second option will be better in this case)

**Review by Nattalli — APPROVED** [review]

**Pull request closed without merging**